From 7f370db4064f8aed5d31b20ab2a9c2afd98afdd6 Mon Sep 17 00:00:00 2001
From: Joe Zheng <joe.zheng@intel.com>
Date: Tue, 17 Nov 2020 05:36:53 -0500
Subject: [PATCH] disable fusectl for AIC

fusectl is a global fs for fuse, it can be used to control all the fuse
connections in the system, e.g. abort the fs, it is OK for the normal
system, but it will be a bad idea for the container use case, if one of
the container abort the fuse fs connection, all the other containers or
the host will be interrupted

the vold in Android R will abort all the fuse connections when stop, it
is horrile for AIC, the fuse fs created by host or aic-manager, e.g.
lxcfs, will be aborted, as well as other ones created by other AIC
instances, what a mess

that is the root cause of the lxcfs abort issue

so for AIC we should disable the fusectl

Change-Id: Iea907ca309e0d785325883b17c16cb59ab526111
Signed-off-by: Joe Zheng <joe.zheng@intel.com>
Tracked-On: ACP-1336
Topic: aic.bug.lxcfs-abort
---
 rootdir/init.rc | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 510d1e911..bf455962c 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -177,9 +177,6 @@ on init
     mount binder binder /dev/binderfs stats=global
     chmod 0755 /dev/binderfs
 
-    # Mount fusectl
-    mount fusectl none /sys/fs/fuse/connections
-
     symlink /dev/binderfs/binder /dev/binder
     symlink /dev/binderfs/hwbinder /dev/hwbinder
     symlink /dev/binderfs/vndbinder /dev/vndbinder
-- 
2.17.1

